### PR TITLE
Embedded audio subtitles support

### PR DESCRIPTION
**Pull request by TRtomasz**
Add support for switching embeded subtitles and audio tracks

### CODE REVIEW

**Review by Ivshti — APPROVED** [review]

File: vlcVideo.android.js [hunk 1]
```diff
@@ -106,7 +133,9 @@ VLCVideo.propTypes = {
     onEndReached: PropTypes.func,
     onError: PropTypes.func,
     onTimeChanged: PropTypes.func,
-    onSeekPerformed: PropTypes.func
+    onSeekPerformed: PropTypes.func,
+    onEmbeddedSubtitlesAvailable: PropTypes.func,
```
Comment (nklhtv) [inline vlcVideo.android.js]: rename to onSubtitlesTracksChanged

File: vlcVideo.android.js [hunk 2]
```diff
@@ -106,7 +133,9 @@ VLCVideo.propTypes = {
...
+    onEmbeddedAudioAvailable: PropTypes.func
```
Comment (nklhtv) [inline vlcVideo.android.js]: rename to onAudioTracksChanged

File: vlcVideo.android.js [hunk 3]
```diff
@@ -67,6 +70,30 @@ class VLCVideo extends Component {
...
+        }
+
+        UIManager.dispatchViewManagerCommand(
+            this._getViewHandle(),
+            UIManager.RCTVLCVideoView.Commands.spu,
+            [index]
+        );
+    }
+
+    audio = (index) => {
```
Comment (nklhtv) [inline vlcVideo.android.js]: rename to setAudioTrack

File: vlcVideo.android.js [hunk 4]
```diff
@@ -67,6 +70,30 @@ class VLCVideo extends Component {
         );
     }
 
+    spu = (index) => {
```
Comment (nklhtv) [inline vlcVideo.android.js]: rename to setSubtitlesTrack

File: android/src/main/java/com/stellarscript/vlcvideo/VLCVideoView.java [hunk 5]
```diff
@@ -316,6 +326,42 @@ private void detachVLCVoutViews() {
...
+
+    public boolean setAudioTrack(int index) {
+        return mMediaPlayer.setAudioTrack(index);
+    }
+
+    public long getSpuDelay() {
+        return mMediaPlayer.getSpuDelay();
+    }
+
+    public boolean setSpuDelay(long delay) {
```
Comment (nklhtv) [inline android/src/main/java/com/stellarscript/vlcvideo/VLCVideoView.java]: remove this function

File: android/src/main/java/com/stellarscript/vlcvideo/VLCVideoView.java [hunk 6]
```diff
@@ -316,6 +326,42 @@ private void detachVLCVoutViews() {
         }
     }
 
+    public TrackDescription [] getAudioTracks() {
+        for(TrackDescription x: mMediaPlayer.getAudioTracks()) {
+        }
+        return mMediaPlayer.getAudioTracks();
+    }
+
+    public TrackDescription [] getSpuTracks() {
```
Comment (nklhtv) [inline android/src/main/java/com/stellarscript/vlcvideo/VLCVideoView.java]: remove this function

File: android/src/main/java/com/stellarscript/vlcvideo/VLCVideoProps.java [hunk 7]
```diff
@@ -23,6 +23,11 @@
     static final int PAUSE_COMMAND_ID = 2;
     static final String SEEK_COMMAND_NAME = "seek";
     static final int SEEK_COMMAND_ID = 3;
+    static final String SET_SPU_COMMAND_NAME = "spu";
+    static final int SET_SPU_COMMAND_ID = 4;
+    static final String SET_AUDIO_COMMAND_NAME = "audio";
+    static final int SET_AUDIO_COMMAND_ID = 5;
     static final int SEEK_COMMAND_TIME_ARGUMENT_INDEX = 0;
```
Comment (nklhtv) [inline android/src/main/java/com/stellarscript/vlcvideo/VLCVideoProps.java:30]: remove those 2 constants. no need of them. hardcode `0` where it is needed. its allways 0

File: android/src/main/java/com/stellarscript/vlcvideo/VLCVideoEventEmitter.java [hunk 8]
```diff
@@ -58,4 +60,24 @@ void emitOnSeekPerformed() {
         mEventEmitter.receiveEvent(mVideoView.getId(), VLCVideoEvents.ON_SEEK_PERFORMED_EVENT, null);
     }
 
+    void emitOnEmbeddedSubtitlesAvailable() {
```
Comment (nklhtv) [inline android/src/main/java/com/stellarscript/vlcvideo/VLCVideoEventEmitter.java]: use subtitlesTracks argument

File: android/src/main/java/com/stellarscript/vlcvideo/VLCVideoEventEmitter.java [hunk 9]
```diff
@@ -58,4 +60,24 @@ void emitOnSeekPerformed() {
...
+        WritableMap event = Arguments.createMap();
+        for(TrackDescription track: mVideoView.getSpuTracks()) {
+            if(!track.name.toLowerCase().contains("disable")) {
+                event.putInt(track.name, track.id);
+            }
+        }
+        mEventEmitter.receiveEvent(mVideoView.getId(), VLCVideoEvents.ON_EMBEDDED_SUBTITLES_AVAILABLE_EVENT, event);
+    }
+
+    void emitOnEmbeddedAudioAvailable() {
```
Comment (nklhtv) [inline android/src/main/java/com/stellarscript/vlcvideo/VLCVideoEventEmitter.java]: use audioTracks argument

**Review by nklhtv — CHANGES_REQUESTED** [review]
use final where it is needed
dont use `embedded` word in this library
use code reformat tool: `if(` must be like `if (`

**Review by nklhtv — APPROVED** [review]

**Pull request merged**